File: src/ChatApp/Services/user-service/src/main/java/com/chatapp/userservice/entity/MessageStatus.java
package com.chatapp.userservice.entity;

import com.chatapp.userservice.entity.keys.MessageStatusId;
import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
@Entity
@Table(name = "message_status")
public class MessageStatus {
    @EmbeddedId
    MessageStatusId id;

    @ManyToOne
    @MapsId("userId")
    @JoinColumn(name = "user_id")
    User user;

    @ManyToOne
    @MapsId("messageId")
    @JoinColumn(name = "message_id")
    Message message;

    @Column(name = "status", columnDefinition = "VARCHAR(255)")
    String status;
}

File: src/ChatApp/Services/Message/ChatApp.Message/Models/MessageStatus.cs
using System;
using System.Collections.Generic;

namespace ChatApp.Message.Models;

public sealed partial class MessageStatus
{
    public string? Status { get; set; }

    public int MessageId { get; set; }

    public int UserId { get; set; }

    public Message Message { get; set; } = null!;

    public User User { get; set; } = null!;
}
